s5_subwords/readme.md

```diff
@@ -1,19 +1,50 @@
+
+### run/run_am.sh
+Start by running the script "run/run_am.sh", it has all the steps related to the acoustic model training. It creates a standard LDA+MLLT+SAT system followed by a TDNN model, with i-vectors and speed perturbation. The script is a standard Kaldi recipe except for the creation of the dictionary as it uses the text corpus to create a grapheme based lexicon instead of a phoneme based model. The "lang" and "dict" used for the acoustic model training are prefixed with "base" to distinguish them from their subword counterparts, which will be created later. The "lang" and "dict" are derived from the text corpus, so no g2p translation is required. Note that the text corpus must be normalized; one sentence per line, all characters should be lower or upper case, and all non-character related signs like numbers or punctuations should be removed. 
+
+### Subword model
+The subword realted steps are in run/run_sw.sh. There are scripts for three subword tokenizers:
+ * [Unigram](https://arxiv.org/abs/1804.10959) 
+ * [Byte Pair Encoding](https://arxiv.org/abs/1508.07909)
+ * [Morfessor](https://morfessor.readthedocs.io/en/latest/)
+ 
+Please read about the tokenizer in their respective repositories, they all perform the same task, but the WER results will differ for each method. The default tokenizer is a "bpe", all scripts for that tokenizer are provided with this repository. To run the tokenizers "unigram" and the "sp_bpe" [Google's Sentence Piece](https://github.com/google/sentencepiece) is needed, a code snippet for a python virtual environment is provided along with a requirments.txt file. 
+
+For each subword tokenization method, there are two parameters, vocabulary size and boundary marking style. The vocabulary size is a parameter that controls how aggressively the words are tokenized, lower values mean fewer and smaller subword units and vice-versa for higher values. The boundary marking styles are the following (the sentence "dog walks" is used as an example): 
+* right-marked (r):  do+ g w+ alk+ s
+* left-marked (l):  do +g w +alk +s
+* left-right-marked (lr): do+ +g w+ +alk+ +s (The default) 
+* word-boundary (wb): do g + w alk s 
+
+The token used to mark the subword boundary can be any sign given that it's not a part of the language, in this setup the default is "+".
+
+### run/run_sw.sh
+The first step in the script is to train the subword tokenization model. Depending on its size, the text corpus or a subset of it is used for that. We have been using 4 million lines of text which should be more than enough. When the tokenizer is trained that model is used to tokenize the entire corpus. This subword tokenized text corpus will be used to prepare the Language model. There are three different subword tokenizers in the script Unigram, 
```

Comment:
Instead of "Depending on its size, the text corpus or a subset of it is used for that." You should put in what you had in the script itself: "If you have a corpus with more than 4 million sentences, create 4 million sentence subset for training." and mention that "This limit is hard coded into the configs for the SP models."

Reply:
Also, you should number each of these steps in the run/run_sw.sh section like 1. 2. 3. 4.